### PR TITLE
test:Add unit tests to jobset success policy functions #491

### PR DESCRIPTION
**Pull request by Dibyanshuaman**

### CODE REVIEW

File: pkg/controllers/success_policy.go [hunk 1]
```diff
@@ -22,6 +22,107 @@ import (
 )
 
 // TODO: add unit tests for the functions in this file.
+// TestJobMatchesSuccessPolicy tests the jobMatchesSuccessPolicy function.
+func TestJobMatchesSuccessPolicy(t *testing.T) {
```
Comment (kannon92) [inline pkg/controllers/success_policy.go:26]: In golang, unit tests are added with a corresponding *_test.go.

So I would move these tests to a pkg/controllers/success_policy_test.go file.

**Comment by Dibyanshuaman** [comment]
So shall i move all the unit test on separate file as instructed by you?

**Comment by Dibyanshuaman** [comment]
No reply was given for my question so no not working right now

**Comment by kannon92** [comment]
I +1 the comment. Yes, please move all the unit test into a separate test file. Your PR is failing due to this. Please make sure the CI jobs are green.

**Comment by Dibyanshuaman** [comment]
As instructed by you i have moved all test case on different file

**Comment by kannon92** [comment]
Thank you @Dibyanshuaman! The CI is failing due to syntax error. Please fix that.

**Comment by danielvegamyhre** [comment]
@Dibyanshuaman please use table-driven tests to test different scenarios for each function. Example: https://github.com/kubernetes-sigs/jobset/blob/8860f832dee2c35d9cb2951260208fcef87cd2db/pkg/controllers/startup_policy_test.go#L25-L45

**Comment by danielvegamyhre** [comment]
@Dibyanshuaman are you still working on this?

**Comment by danielvegamyhre** [comment]
Closing due to inactivity

/close

**Pull request closed without merging**